sd-2019-stack-overflow-frontend/src/view/SmartAnswerList.js

```diff
@@ -0,0 +1,98 @@
+import React, {Component} from "react";
+import questionModel from "../model/questionModel";
+import answerModel from "../model/answerModel";
+import questionListPresenter from "../presenter/questionListPresenter";
+import answerListPresenter from "../presenter/answerListPresenter";
+import SmartNavbar from "./SmartNavbar";
+import Footer from "./Footer";
+import AnswerList from "./AnswerList";
+import AskQuestionModal from "./AskQuestionModal";
+import EditAnswerModal from "./EditAnswerModal";
+
+const mapModelStateToComponentState = (questionModelState, answerModelState, props) => (
+    {
+        question: questionModelState.questions.filter(q => q.id === Number(props.match.params.questionId))[0],
+        answers: answerModelState.answers.filter(a => a.questionId === Number(props.match.params.questionId))
+            .sort(function(a1, a2) {return a2.voteScore - a1.voteScore;}),
+        newAnswer: answerModelState.newAnswer,
+        askQuestionModalActive: questionModelState.askQuestionModalActive,
+        newQuestion: questionModelState.newQuestion,
+        newTag: questionModelState.newTag,
+        editAnswerModalActive: answerModelState.editAnswerModalActive,
+        editedAnswer: answerModelState.editedAnswer
+    }
+);
+
+export default class SmartAnswerList extends Component {
+    constructor(props) {
+        super(props);
+        this.state = mapModelStateToComponentState(questionModel.state, answerModel.state, this.props);
+        this.listener = () => this.setState(mapModelStateToComponentState(questionModel.state, answerModel.state, this.props));
+        answerModel.addListener("AnswerModelChange", this.listener);
+        questionModel.addListener("QuestionModelChange", this.listener);
+    }
+
+    componentWillUnmount() {
+        answerModel.removeListener("AnswerModelChange", this.listener);
+        questionModel.removeListener("QuestionModelChange", this.listener);
+    }
+
+    componentDidUpdate(prev) {
+        if (prev.match.params.questionId !== this.props.match.params.questionId) {
+            this.setState(mapModelStateToComponentState(questionModel.state, answerModel.state, this.props));
+        }
+    }
+
+    render() {
+        return (
+            <div>
+                <SmartNavbar />
+                {
+                    this.state.askQuestionModalActive === true
```

Comment:
Small hint: in JS you can just do `this.state.askQuestionModalActive ? ` instead of `this.state.askQuestionModalActive === true ?`